### ISSUE TITLE
Performance losses because of unnecessary stack trace capturing in LoggerImpl.Write method

### ISSUE THREAD

**Issue by vital-bit-vlad** [NLog/NLog#2717]
**Type** (choose one):
- Feature request

**NLog version**: 4.5.4

**Platform**:  .Net 4.5

**Current NLog config** (xml or C#, if relevant)
Doesn't matter

If I use API like this:
```c#
_logger
    .Debug()
    .Message(message)
    .Write(callerMemberName, callerFilePath, callerLineNumber);
```

I've already provided caller info, why do you still capture stack trace?

```c#
var stackFrames = stackTrace.GetFrames();
int? firstUserFrame = FindCallingMethodOnStackTrace(stackFrames, loggerType);
int? firstLegacyUserFrame = firstUserFrame.HasValue ? SkipToUserStackFrameLegacy(stackFrames, firstUserFrame.Value) : (int?)null;
logEvent.GetCallSiteInformationInternal().SetStackTrace(stackTrace, firstUserFrame ?? 0, firstLegacyUserFrame);

```

It causes significant performance losses for no reason - 84% of logging time is wasted:
![image](https://user-images.githubusercontent.com/33297215/39948886-fe424702-5545-11e8-9d04-b09b1a80760b.png)

**Comment by snakefoot** [comment]
This is resolved with #2527. But it includes some breaking changes.

**Comment by 304NotModified** [comment]
This is only where there is a `${callsite}` in your message, isn;t?

**Comment by snakefoot** [comment]
Resolved by #3431 + #3975 for NLog 5.0
